src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -244,14 +245,23 @@ public static CompositeConfigurationSource CreateDefaultConfigurationSource()
 
         internal bool IsIntegrationEnabled(string name)
         {
-            bool disabled = Integrations[name].Enabled == false || DisabledIntegrationNames.Contains(name);
-            return TraceEnabled && !disabled;
+            if (TraceEnabled && !DomainMetadata.ShouldAvoidAppDomain())
+            {
+                return Integrations[name].Enabled == false || DisabledIntegrationNames.Contains(name);
```

Comment:
With the changes, this is now returning `disabled` instead of `!disabled`.

```suggestion
                bool disabled = Integrations[name].Enabled == false || DisabledIntegrationNames.Contains(name);
                return !disabled;
```